public/pages/CreateMonitor/containers/CreateMonitor/utils/monitorToFormik.js

```diff
@@ -28,10 +29,37 @@ export default function monitorToFormik(monitor) {
     inputs,
     ui_metadata: { schedule = {}, search = {} } = {},
   } = monitor;
+
   // Default searchType to query, because if there is no ui_metadata or search then it was created through API or overwritten by API
   // In that case we don't want to guess on the UI what selections a user made, so we will default to just showing the extraction query
-  const { searchType = 'query', fieldName } = search;
-  const isAD = searchType === SEARCH_TYPE.AD;
+  let { searchType = 'query', fieldName } = search;
+  // There is a new searchType - http, and can be created through API, this condition needs to be handled.
+  if (_.isEmpty(search) && 'http' in inputs[0]) searchType = SEARCH_TYPE.HTTP;
+
+  function inputsToFormik() {
```

Comment:
I added a new function to deal with different return values for different input types to improve the readability.